doc/arch/adr-005-api-client-app-auth.md

```diff
@@ -0,0 +1,108 @@
+easy to uncover and then fake by anyone trying to get un-checked access to the
+API. Our solution should:
+
+- Allow us to easily update the client "secret" so that whenever someone steals it, we can
+just cycle it out to a new one. (Or this could happen at some regular interval)
+- Not be overly complicated/difficult for us when we know there's no truly secure
+solution (barring doing all our rendering server-side)
+
+Some possibilities are:
+
+1. Give each environment of the client app its own API token with which it
+would make guest user requests
+   - Keeps a single system for API authentication
+   
+1. Set a special flag in the session token that the API would check for in
+unauthenticated requests
+   - Could be the CSRF token or something else. Might involve some custom
+   generation/management
+   
+1. Some combination of (1) and (2)
+   - Things could get a lot more complex for us, but not that much more complex
+   for any would-be attacker
+
+## Decision
+
+**Give the client app its own API token**
+
+Setting up the app with its own API token will allow us to both
+- cycle the key easily because Django Rest Framework (DRF) handles token
+generation/re-generation, and we already use it for "real" API users
+- keep the auth system simple: everyone uses a token for auth
+
+Starting with a client API token as a base, we can always add additional checks on
+top as enhancements down the road.
+
+##### How the client app should get the token
+
+If we hard-code the token on the frontend (or anywhere), we'll have to deploy any time we need
+to cycle the token. We should instead pass the token to the client app via `clientSettings`. 
+
+##### How the server should get the token
+
+To get the token to put in the client's settings, the server can look up the
+token via the client app's `user_id`. The `user_id` will have to remain
+secret so that there's no programmatic way get the token outside of swiping
+it off the client app.
+
+## Consequences
+
+#### Backdoor to the API
+As discussed in Context, allowing the client app to make requests without a user creates a 
+backdoor into the API. There's no sensitive data available via the API, but we (1) don't want 
+people to make unlimited, resource-intensive requests unchecked, and (2) don't want to lose
+the potential to charge users. 
+
+Throttling could help with (1), cycling the client app token quickly/frequently could help with
+(2).
+
+
+#### More Complex Throttling
+
+We want to throttle users of the API, and if the client app is just another, regular user
+of the API with a token it would get throttled the same. All guest MMW users in the world
+would share a cumalitive number of requests per minute.
+
+To fix this we'll need to treat the client app's token as a special case for throttling: 
+
+if token is client app:
+
+cache number of requests per IP (like [AnonRateThrottle](http://www.django-rest-framework.org/api-guide/throttling/#anonratethrottle))
```

Comment:
For this, we just need to make sure the IP it uses is truly the clients vs. the IP of our load balancer, or some other hop internal to our infrastructure. May be hard to test outside of EC2.